### PR TITLE
Add size prop to checkbox

### LINKED ISSUE #799: Checkbox label doesn't allow font size small
**Describe the bug**
Checkbox label doesn't allow font size small (14pt). It should allow it because it makes sense in some scenarios, such as when regular/small inputs with font size small are already being used in a form.

**To Reproduce**
Open the styleguide documentation and see that this isn't possible in the current implementation.

**Expected behavior**
The component should allow small font size instead of body whenever needed.

**Screenshots**
Group 13

### PR DESCRIPTION
**Pull request by arthurepc**
#### What is the purpose of this pull request?

Fix #799 

#### How should this be manually tested?

Run this project locally and go to the Checkbox page. Make sure that when the `size` prop is passed, the text size changes.

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+
+- size prop to checkbox.
```
Comment (bivillar) [inline CHANGELOG.md]:
```suggestion
- Size prop to checkbox.
```

File: react/components/Checkbox/index.js [hunk 2]
```diff
@@ -119,6 +121,8 @@ Checkbox.propTypes = {
   value: PropTypes.string,
   /** Partial state */
   partial: PropTypes.bool,
+  /** Label size */
+  size: PropTypes.oneOf(['small']),
```
Comment (bivillar) [inline react/components/Checkbox/index.js]: Wouldn't it be better if this was a boolean prop named `small`?

Comment (kevinch) [reply react/components/Checkbox/index.js]: Not sure @bivillar since in the future we could add a "regular" or "large" for x reasons.

Comment (emersonlaurentino) [reply react/components/Checkbox/index.js]: I agree with what @kevinch said

**Pull request merged**